tools/lemmacmd.go

```diff
@@ -0,0 +1,267 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/pbkdf2"
+
+	"github.com/mailgun/lemma/random"
+	"github.com/mailgun/lemma/secret"
+)
+
+var (
+	inputpath  = flag.String("in", "", "path to file to be read in")
+	outputpath = flag.String("out", "", "path to file to be written out")
+	mode       = flag.String("mode", "", "mode: enc[rypt] or dec[rypt]")
+	keypath    = flag.String("keypath", "", "path to base64-encoded 32-byte key on disk, if no path is provided, a passphrase will be used")
+	itercount  = flag.Int("itercount", 524288, "if a passphrase is used, iteration count for pbkdf#2")
```

Comment:
What do you think of using a subcommand instead of a mode flag?

That is, instead of `lemmacmd -mode=enc ...` you type `lemmacmd encrypt ...`